packages/mutation-testing-elements/src/components/svg/index.ts

```diff
@@ -1,10 +1,47 @@
+  private readonly typescript: SVGTemplateResult = svg`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M23.827,8.243A4.424,4.424,0,0,1,26.05,9.524a5.853,5.853,0,0,1,.852,1.143c.011.045-1.534,1.083-2.471,1.662-.034.023-.169-.124-.322-.35a2.014,2.014,0,0,0-1.67-1c-1.077-.074-1.771.49-1.766,1.433a1.3,1.3,0,0,0,.153.666c.237.49.677.784,2.059,1.383,2.544,1.095,3.636,1.817,4.31,2.843a5.158,5.158,0,0,1,.416,4.333,4.764,4.764,0,0,1-3.932,2.815,10.9,10.9,0,0,1-2.708-.028,6.531,6.531,0,0,1-3.616-1.884,6.278,6.278,0,0,1-.926-1.371,2.655,2.655,0,0,1,.327-.208c.158-.09.756-.434,1.32-.761L19.1,19.6l.214.312a4.771,4.771,0,0,0,1.35,1.292,3.3,3.3,0,0,0,3.458-.175,1.545,1.545,0,0,0,.2-1.974c-.276-.395-.84-.727-2.443-1.422a8.8,8.8,0,0,1-3.349-2.055,4.687,4.687,0,0,1-.976-1.777,7.116,7.116,0,0,1-.062-2.268,4.332,4.332,0,0,1,3.644-3.374A9,9,0,0,1,23.827,8.243ZM15.484,9.726l.011,1.454h-4.63V24.328H7.6V11.183H2.97V9.755A13.986,13.986,0,0,1,3.01,8.289c.017-.023,2.832-.034,6.245-.028l6.211.017Z" style="fill:#007acc"/></svg>`;
+
+  private readonly svgMapping: Map<string, SVGTemplateResult> = new Map();
+
+  constructor() {
+    this.svgMapping.set('cs', this.csharp);
+    this.svgMapping.set('html', this.html);
+    this.svgMapping.set('java', this.java);
+    this.svgMapping.set('js', this.js);
+    this.svgMapping.set('scala', this.scala);
+    this.svgMapping.set('ts', this.typescript);
+  }
+
+  public getIconForFile(fileName: string): SVGTemplateResult {
+    const fileExtension = this.getFileExtension(fileName);
+    const svg = this.svgMapping.get(fileExtension);
+
+    if (svg !== undefined) {
+      return svg;
+    } else {
+      return this.file;
+    }
+  }
+
+  public getIconForFolder(): SVGTemplateResult {
+    return this.directory;
+  }
+
+  private getFileExtension(fileName: string) {
+    const tokens = fileName.split('.');
+    return tokens[tokens.length - 1];
```

Comment:
Why not `return fileName.split('.').pop()`. I thought you liked functional programming? ;)

Reply:
I DO like functional programming but I also like that my state does not get mutated like `pop()` does ;)

```ts
const data = [1, 2, 3, 4]

console.log(data) // [1, 2, 3, 4]
console.log(data.pop()) // 4
console.log(data) // [1, 2, 3]
```